src/deadline/client/_pid_utils.py

```diff
@@ -1,85 +1,196 @@
+    return True
+
+
+def release_pid_lock(
+    pid_file_full_path: str, print_function_callback: Callable[[str], None]
+) -> bool:
     """
-    Obtains a lock on the pid file by writing the current process id to the file.
-    :param logger:
-    :param pid_file_full_path:
-    :return: boolean, True if pid lock was obtained successfully
+    Releases the pid lock by deleting the pid file.
+    :param pid_file_full_path: full path of the pid lock file
+    :param print_function_callback: print_function_callback (Callable str -> None, optional): Callback to print messages produced in this function.
+                Used in the CLI to print to stdout using click.echo. By default, ignores messages.
+    :return: boolean, True if pid lock released successfully
     """
+    print_function_callback(f"Releasing pid lock at {pid_file_full_path}")
+
+    # Get the current process's id to obtain lock
+    current_process_pid: int = os.getpid()
+
+    # Check if pid lock file does not exist. Returns True if file doesn't exist as there is no lock to be released.
+    if not os.path.exists(pid_file_full_path):
+        print_function_callback(f"Pid lock file does not exist at {pid_file_full_path}")
+        return True
+
+    # Try to open pid file at download progress location in read mode
+    with open(pid_file_full_path, "r") as f:
+        # Read pid file and obtain the process id from file contents
+        pid = f.read()
+
+    # Process pid from file is same as current process pid - release pid lock
+    if int(pid) == current_process_pid:
+        print_function_callback(
+            f"Process with pid {pid} is the current process. Deleting pid file."
+        )
+        os.remove(pid_file_full_path)
+        return True
+    # Process pid from file is different from current process pid - do not release lock
+    else:
+        print_function_callback(
+            f"Process with pid {pid} is not the current process. Skipping pid file deletion."
+        )
+        return False
+
+
+def _lock_pid_file_and_release_dangling_pid_lock(pid: str, pid_file_full_path: str) -> bool:
+    """
+    Helper method to lock the pid file and delete it to release a dangling pid lock for a pid which is not active
+    The primitive lock is obtained on the pid file so concurrent processes cant delete it
+    This would be caught in race conditions if primitive file locking is disabled on a customer's machine.
+    :param pid: inactive pid in the file
+    :param pid_file_full_path: full path of the pid file
+    :return: returns True if it was able to release the inactive pid dangling lock.
+        Throws a runtime exception if the pid was updated by a different concurrent process and does not try to delete file
+        Throws an unexpected exception if we get into an error while trying to obtain lock on file.
+    """
+    file_size: int = os.path.getsize(os.path.realpath(pid_file_full_path))
+    file_locked_for_delete = open(pid_file_full_path, "r+")
```

Comment:
This should be `with open(pid_file_full_path, "r+") as file_locked_for_delete:`. Did you try this and run into any difficulties with the locking?